lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1609,6 +1609,14 @@ LogicalResult FIRRTLLowering::visitDecl(NodeOp op) {
       // do not attach a symbol name.
       auto wire = builder.create<sv::WireOp>(operand.getType(), name);
       builder.create<sv::ConnectOp>(wire, operand);
+
+      if (AnnotationSet(op).hasDontTouch()) {
+        auto moduleName = cast<hw::HWModuleOp>(op->getParentOp()).getName();
+        auto symName = builder.getStringAttr(
+            (Twine("__") + moduleName + Twine("__") + name.getValue())
+                .str());
+        wire->setAttr("sym_name", symName);
+      }
```

Comment:
This looks correct to me. 

Shouldn't this also work for nodes that don't have a name?

Reply:
Do we need to rely on whether a name is present or not?
Can we just lower it to wires, only if DontTouch annotation is present ?